blues/simulation.py

```diff
@@ -933,7 +945,27 @@ def _accept_reject_move_(self, write_move=False):
 
             #If reject move, reset positions in ncmc context to before move
             md_state0 = self.stateTable['md']['state0']
-            self._ncmc_sim.context = self.setContextFromState(self._ncmc_sim.context, md_state0)
+            self._ncmc_sim.context = self.setContextFromState(self._ncmc_sim.context, md_state0, velocities=False)
+
+            # Check potential energy is that of the last MD step
+            md_PE = self._md_sim.context.getState(getEnergy=True).getPotentialEnergy()
+            if not math.isclose(md_state0['potential_energy']._value, md_PE._value, rel_tol=float('1e-%s' % rtol)):
+                logger.error('Last MD potential energy %s != Current MD potential energy %s. Could not set the MD simulation to the previous state.' %(md_state0['potential_energy'], md_PE))
```

Comment:
"Could not set the MD simulation to..." may not be the right error message; I would want it to say that it's rejected an NCMC move and so the potential energy SHOULD match that of the prior state, but that it found that it doesn't, so it is terminating because this means something is wrong.
"Could not set" sort of sounds like there is an error with the setting process, whereas this is just a check to see if something deeper has gone wrong.